### DISCUSSION TITLE
formState.isSubmitted set true but form has validation errors and was not passed to submit handler.

### DISCUSSION THREAD

**Discussion by jjwwiodyne** [react-hook-form/react-hook-form#2793] [Q&A]
I'm not sure if I'm misunderstanding `formState.isSubmitted`. When the form is submitted, there are validation errors that get picked up, and the form submission handler doesn't fire but `formState.isSubmitted` is set to true, even though `errors` has content. Is `isSubmitted` considered true whether it was actually sent to the submission handler or not?

```
import React, { useState, useEffect, SyntheticEvent } from 'react';
import { useDispatch, useSelector } from "../../../datastore/store";
import { IOFormInput } from "../../common-components/form-parts"
import ReactDOM from 'react-dom';
import { useTranslation } from 'react-i18next';
import { SupportTicketUpdateType } from "../../../models/accounts"
import { useForm } from 'react-hook-form'
import i18n from "../../../i18n"

// Form Definition
export interface TicketInputFormProps {
  handleSubmit: (values:any) => void
}

function FormComponent(props:any) {
  const last_ticket_created:SupportTicketType = useSelector(state => { return state.helpReducer.last_ticket_created });
  const dispatch = useDispatch();
  const { register, errors, handleSubmit, formState } = useForm({
    mode: "onSubmit",
  });
  
  async function handleTicketSubmit(values:SupportTicketUpdateType) {
    console.log("hi");
  }
  
  console.log(formState.isSubmitted);
  console.log(errors);

  return <React.Fragment>
    {formState.isSubmitting && <div className="spinner">
      <span className="ticket_progress_label">Creating ticket...</span>
      </div>}
    {last_ticket_created && <span className="ticket_progress_label">{t("help.create_ticket.ticket_created_msg", {ticket_no: last_ticket_created.ticket_no})}</span>}
    {!formState.isSubmitting  && <form onSubmit={handleSubmit(handleTicketSubmit)}>
      {errors && <div className="ticket_creation_error">ERRORS GO HERE</div>}
      <IOFormInput
        name="subject"
        type="text"
        label={t("help.create_ticket.fields.subject.prompt")}
        forwardRef={register({required: true})}
        validationErrorMsg={errors.subject && i18n.t("help.create_ticket.fields.subject.errors.default")}
        placeholder={t("help.create_ticket.fields.subject.placeholder")}/>
      </div>
      <div className="ticket_create_actions">
        <IOFormButton type="submit" label={t("help.create_ticket.buttons.create_ticket")}/>
      </div>
  </form>}
  </React.Fragment>
}

export default FormComponent;
```

**Comment by bluebill1049** [comment]
I think we have the following options.

1. change `isSubmitted` to only for valid submit, right now submit count and isSubmitted both consider form submit as a submit which is a generic submit count and whether submitted. (This is a breaking change, which i personally prefer to avoid)
2. create a new form state called `isSuccessfullySubmitted`
3. main a piece of a state in your app, which is probably ideally then you can invalid or valid form is successfully submitted.

**Reply by fachrur95** [reply]
how about submitCount sir? when form error, counter is still counted how many clicked.

**Comment by jjwwiodyne** [comment]
Semantically, changing it to be set only for a valid submit seems the most
elegant since it's implied the submit failed if isSubmitting changed to
false and isSubmitted is also still false. But I bet there's code that
depends on isSubmitted=true meaning the submit button was clicked. So
having success and failure flags for submission seems least likely to break
things.

-J
